compiler/rustc_lint/src/unused.rs

```diff
@@ -161,7 +161,18 @@ impl<'tcx> LateLintPass<'tcx> for UnusedResults {
 
         if let Some(must_use_op) = must_use_op {
             cx.struct_span_lint(UNUSED_MUST_USE, expr.span, |lint| {
-                lint.build(&format!("unused {} that must be used", must_use_op)).emit()
+                let mut lint = lint.build(&format!("unused {} that must be used", must_use_op));
+                lint.note(&format!("the {} produces a value", must_use_op));
+                if let Ok(snippet) = cx.sess().source_map().span_to_snippet(expr.span) {
+                    lint.span_suggestion(
+                        expr.span,
+                        "use `let _ = ...` to ignore it",
+                        format!("let _ = {}", snippet),
+                        Applicability::MachineApplicable,
+                    )
+                    .emit()
+                }
```

Comment:
```suggestion
                lint.span_label(expr.span, &format!("the {} produces a value", must_use_op));
                lint.span_suggestion_verbose(
                    expr.span,
                    "use `let _ = ...` to ignore the resulting value",
                    "let _ = ".to_string(),
                    Applicability::MachineApplicable,
                );
```

Reply:
Thanks! Applied it with changing `expr.span` to `expr.span.shrink_to_lo()`.